### ISSUE TITLE
Unbalanced curly brackets

### ISSUE THREAD

**Issue by ileitch** [nicklockwood/SwiftFormat#411]
Give the following code:

```swift
Array([]
    .map {
        ""
    }.something())
```

SwiftFormat will format it as:

```swift
Array([]
    .map {
        ""
}.something())
```

It looks as though in an attempt to balance the parenthesis, it has ignored the curly bracket. I think SwiftFormat should give precedence to the inner most bracket/parenthesis, or break the parenthesis onto a new line.

**Comment by bryanforbes** [comment]
I have a similar scenario:

```swift
class Thing {
    private let somePublisher = Publishers.Factory<OtherThing, Never> { dispatcher in
        // do some stuff here
        return AnyCancellable {}
    }
    .share(replay: 1)
    .eraseToAnyPublisher()
}
```

Gets formatted as:

```swift
class Thing {
    private let somePublisher =
        Publishers.Factory<OtherThing, Never> { dispatcher in
        // do some stuff here
        return AnyCancellable {}
    }
    .share(replay: 1)
    .eraseToAnyPublisher()
}
```

And I was expecting it to format as:

```swift
class Thing {
    private let somePublisher =
        Publishers.Factory<OtherThing, Never> { dispatcher in
            // do some stuff here
            return AnyCancellable {}
        }
        .share(replay: 1)
        .eraseToAnyPublisher()
}
```

**Comment by nicklockwood** [comment]
@ileitch this is fixed in 0.45.0.

@bryanforbes you example is not yet fixed, as trailing closure indenting needs a different fix.

**Comment by nicklockwood** [comment]
@bryanforbes your example is now also working correctly in 0.45.3.
